### PR TITLE
fix: add theme.original space to save colors

### PR DESCRIPTION
**Pull request by atanasster**
issue #1406
added tests
added field theme.original.colors to save colors in emotion theme object

### CODE REVIEW

File: packages/css/src/types.ts [hunk 1]
```diff
@@ -808,4 +808,14 @@ export interface Theme {
    * @see https://theme-ui.com/components/message#variants
    */
   messages?: Record<string, ThemeUIStyleObject>
+
+  /**
+   * When the theme values are replaced with css custom vars, save original values
+   */
+  original?: {
```
Comment (hasparus) [inline packages/css/src/types.ts]: I'm skeptical towards including it in the types for our public API.
1. Users won't ever add it manually. If they have a key called original, we'd overwrite it.
2. We don't have that value on the theme in **our context**. We add it to theme's shallow copy passed to **Emotion's context**.

File: packages/color-modes/src/index.tsx [hunk 2]
```diff
@@ -131,8 +131,10 @@ export const ColorModeProvider: React.FC = ({ children }) => {
 
   if (theme.useCustomProperties !== false) {
     emotionTheme.colors = toCustomProperties(emotionTheme.colors, 'colors')
+    emotionTheme.original = {
```
Comment (hasparus) [inline packages/color-modes/src/index.tsx]: There's a tiny risk that users will have a key named "original" in their themes, what makes it a breaking change.
A use-case for a Symbol perhaps? We could keep this more "private".

File: packages/css/src/types.ts (hunk 1 above)
Comment (atanasster) [reply packages/css/src/types.ts]: removed

**Comment by hasparus** [comment]
Awesome. Looks good to merge.

**Review by lachlanjc — APPROVED** [review]

**Comment by hasparus** [comment]
Okay, after fighting with color mode flash for _ages_ and reading https://github.com/system-ui/theme-ui/pull/259 for a tenth time I'm positive that we need to bring back `rawColors`
-  either by saving them to `__original.colors` like this PR 
- **or** saving transformed colors to `theme.__colorsVars`, and changing scales to use `__colorsVars` instead of `colors`.
  - This way we'd leave `colors` intact and avoid crashing @LekoArts's Cypress tests again :P 
  - Leaving `colors` intact is cool in case users want to transform them with functions from polished / pass them to threejs canvas or anywhere else.

**Comment by atanasster** [comment]
I dont see any reference to ‘rawColors’ in this PR?

**Comment by hasparus** [comment]
> I dont see any reference to ‘rawColors’ in this PR?

Sorry for the confusion.
`__original.colors` already existed in Theme UI in the past and was called `rawColors`. Debugging color mode flash led me to the CSS properties and the observation that sometimes we read colors from Emotion context, and sometimes (e.g. code snippets in Theme UI docs) we read them unprocessed from our context.

I'm in the process of fixing merge conflicts and merging this PR right now — The alternative to `theme.original.colors` - saving processed css properties to `theme.colorVars` is infeasible due to how our packages are separated.

**Pull request merged**